src/ack_test_infra/k8s/resource.py

```diff
@@ -0,0 +1,381 @@
+    return spec
+
+def create_reference(crd_group: str,
+                     crd_version: str,
+                     resource_plural: str,
+                     resource_name: str,
+                     namespace: str):
+    """
+    Create an instance of CustomResourceReference based on the parameters
+
+    :param crd_group: CRD Group
+    :param crd_version: CRD version
+    :param resource_plural: resource plural
+    :param resource_name: name of resource to be created in cluster
+    :param namespace: namespace in which resource should be created
+
+    :return: an instance of CustomResourceReference
+    """
+    reference = CustomResourceReference(
+        crd_group, crd_version, resource_plural, resource_name, namespace=namespace
+    )
+    return reference
+
+def create_resource(reference: CustomResourceReference,
+                    spec: object):
+    """
+    Create a resource from the reference and wait to be consumed by controller
+    
+    :param reference: instance of CustomResourceReference which needs to be created
+    :param spec: spec of the resource corresponding to the reference
+
+    :return: resource if it was created successfully, otherwise None
+    """
+    resource = create_custom_resource(reference, spec)
+    resource = wait_resource_consumed_by_controller(reference)
+    return resource
+
+def load_and_create_resource(service_name: str,
+                             crd_group: str,
+                             crd_version: str,
+                             resource_plural: str,
+                             resource_name: str,
+                             spec_file_name: str,
+                             replacements: object,
+                             namespace: str = "default"):
+    """
+    Helper method to encapsulate the common methods used to create a resource.
+    Load a spec file from disk, create an instance of CustomResourceReference and resource in K8s cluster.
+    See respective methods for paramater definitions and return types
+
+    :returns: an instance of CustomResourceReference, spec loaded from disk, resource created from the reference
+    """
+    spec = load_resource(service_name, spec_file_name, replacements)
+    reference = create_reference(crd_group, crd_version, resource_plural, resource_name, namespace)
+    resource = create_resource(reference, spec)
+    return reference, spec, resource
+
+def _get_k8s_api_client() -> ApiClient:
+    global _k8s_api_client
+    if _k8s_api_client is None:
+        _k8s_api_client = config.new_client_from_config()
+    return _k8s_api_client
+
+
+def create_k8s_namespace(namespace_name: str):
+    _api_client = _get_k8s_api_client()
+    return client.CoreV1Api(_api_client).create_namespace(
+        client.V1Namespace(name=namespace_name))
+
+
+def delete_k8s_namespace(namespace_name: str):
+    _api_client = _get_k8s_api_client()
+    return client.CoreV1Api(_api_client).delete_namespace(namespace_name)
```

Comment:
How about placing these in a separate `namespace.py` module and calling them just `create` and `delete`?

Then, callers could do:

```python
from acktest.k8s import namespace

namespace.create(...)

namespace.delete(...)
```
...

Reply:
We could take the same approach for secrets...

```python
from acktest.k8s import secret

secret.create_opaque(...)

secret.delete(...)

Reply:
This is a more object oriented, fashion? Do we want to do this with each of the types?